microsoft/testsuites/xfstests/xfstesting.py

```diff
@@ -0,0 +1,291 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from typing import List
+
+from lisa import (
+    Environment,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    schema,
+    simple_requirement,
+)
+from lisa.features import Nvme
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.tools import Fdisk, Mkfs, Mount, Parted, Xfstests
+from lisa.tools.mkfs import FileSystem
+from lisa.util import find_patterns_in_lines
+
+_scratch_folder = "/root/scratch"
+_test_folder = "/root/test"
+
+
+def _configure_disk(
+    node: Node,
+    disk_name: str,
+    first_disk: str,
+    second_disk: str,
+    first_mountpoint: str = _test_folder,
+    second_mountpoint: str = _scratch_folder,
+    file_system: FileSystem = FileSystem.xfs,
+) -> None:
+    mount = node.tools[Mount]
+    fdisk = node.tools[Fdisk]
+    parted = node.tools[Parted]
+    mkfs = node.tools[Mkfs]
+
+    node.execute(f"rm -r {first_mountpoint}", sudo=True)
+    node.execute(f"rm -r {second_mountpoint}", sudo=True)
+
+    fdisk.delete_partitions(disk_name)
+
+    mount.umount(first_disk, first_mountpoint)
+    mount.umount(second_disk, second_mountpoint)
+
+    parted.make_label(disk_name)
+    parted.make_partition(disk_name, "primary", "1", "50%")
+    parted.make_partition(disk_name, "secondary", "50%", "100%")
+
+    mkfs.format_disk(first_disk, file_system)
+    mkfs.format_disk(second_disk, file_system)
+
+    node.execute(f"mkdir {first_mountpoint}", sudo=True)
+    node.execute(f"mkdir {second_mountpoint}", sudo=True)
+
+
+def _get_data_disks(environment: Environment, node: Node) -> List[str]:
+    if isinstance(environment.platform, AzurePlatform):
+        pattern = re.compile(r"/dev/disk/azure/scsi[1-9]/lun[0-9][0-9]?", re.M)
+        cmd_result = node.execute(
+            "ls -d /dev/disk/azure/scsi*/*", shell=True, sudo=True
+        )
+        matched = find_patterns_in_lines(cmd_result.stdout, [pattern])
+        assert matched[0]
+        lun_disk_array: List[str] = [""] * len(matched[0])
+        for disk in matched[0]:
+            cmd_result = node.execute(f"readlink -f {disk}", shell=True, sudo=True)
+            lun_disk_array[
+                int(disk.split("/")[-1].replace("lun", ""))
+            ] = cmd_result.stdout
+    return lun_disk_array
+
+
+@TestSuiteMetadata(
+    area="storage",
+    category="community",
+    description="""
+    This test suite is to validate different types of data disk on Linux VM
+     using xfstests.
+    """,
+)
+class Xfstesting(TestSuite):
+    TIME_OUT = 7200
```

Comment:
add comments to explain why it needs longer time.

Reply:
fixed